flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/wordcount/WordCount.java

```diff
@@ -99,7 +118,28 @@ public static void main(String[] args) throws Exception {
 	 * FlatMapFunction. The function takes a line (String) and splits it into
 	 * multiple pairs in the form of "(word,1)" ({@code Tuple2<String, Integer>}).
 	 */
-	public static final class Tokenizer implements FlatMapFunction<String, Tuple2<String, Integer>> {
+	public static final class Tokenizer extends RichFlatMapFunction<String, Tuple2<String, Integer>> {
+
+		private final Collection<String> inputFiles;
+
+		public Tokenizer() {
+			this(null);
+		}
+
+		public Tokenizer(@Nullable Collection<String> inputFiles) {
+			this.inputFiles = inputFiles;
+		}
+
+		@Override
+		public void open(Configuration conf) throws IOException {
+			if (inputFiles != null && inputFiles.size() > 0) {
+				List<Object> classCollection = new ArrayList<>(inputFiles.size());
+				for (String input : inputFiles) {
+					classCollection.add(FileSystem.getUnguardedFileSystem(URI.create(input)));
+				}
+				PluginUtils.checkClassIsolationInDifferentPlugins(classCollection);
```

Comment:
Those `checkClassIsolationInDifferentPlugins ` (both in batch and streaming `WordCount`) checks  shouldn't happen in the examples, but in the dummy file system implementations. Otherwise they will be confusing people reading those code examples.

Reply:
You concern is reasonable. I will move the class isolation check to dummy file system Implementations.